### PR TITLE
docs: term clarifications in md files

### PR DESCRIPTION
**Pull request by vonovak**
### TL;DR

Various documentation files are updated for better alignment with the current state of the Expo environment. Terminology is refined, redundant details are removed, and references are updated.

# Why
- To reflect current practices and tools, minimize any confusion with outdated terminology, and improve overall documentation quality.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by vonovak** [comment]
* **#30927** <a href="https://app.graphite.dev/github/pr/expo/expo/30927?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#30926** <a href="https://app.graphite.dev/github/pr/expo/expo/30926?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @vonovak and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Manual smoke tests are included in `apps/native-component-list`, this is a good
 
 > 💡 How does `bare-expo` relate to `test-suite`?
 >
-> `bare-expo` is a bare workflow app that links all of the Expo SDK dependencies in the `packages/` directory in order to be able to run projects in the `apps/` directory in the bare workflow rather than the Expo client. It currently only runs `test-suite`. `test-suite` is a regular managed workflow Expo app with some custom code to turn it into a test runner. If you run `expo start` in the `test-suite` directory you can load the project in Expo client. `bare-expo` imports the `test-suite` app root component and uses it as its own root component.
+> `bare-expo` is a bare RN app that links all of the Expo SDK dependencies in the `packages/` directory in order to be able to run projects in the `apps/` directory. It currently only runs `test-suite` which is an Expo app with some custom code to turn it into a test runner. If you run `expo start` in the `test-suite` directory you can load the project in Expo Go. `bare-expo` imports the `test-suite` app root component and uses it as its own root component.
```
Comment (Simek) [inline CONTRIBUTING.md:25]:
```suggestion
> `bare-expo` is a bare React Native app that links all of the Expo SDK dependencies in the `packages/` directory in order to be able to run projects in the `apps/` directory. It currently only runs `test-suite` which is an Expo app with some custom code to turn it into a test runner. If you run `expo start` in the `test-suite` directory you can load the project in Expo Go. `bare-expo` imports the `test-suite` app root component and uses it as its own root component.
```

Nit, avoid abbreviations.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ Manual smoke tests are included in `apps/native-component-list`, this is a good
 
 > 💡 How does `bare-expo` relate to `test-suite`?
 >
-> `bare-expo` is a bare workflow app that links all of the Expo SDK dependencies in the `packages/` directory in order to be able to run projects in the `apps/` directory in the bare workflow rather than the Expo client. It currently only runs `test-suite`. `test-suite` is a regular managed workflow Expo app with some custom code to turn it into a test runner. If you run `expo start` in the `test-suite` directory you can load the project in Expo client. `bare-expo` imports the `test-suite` app root component and uses it as its own root component.
+> `bare-expo` is a bare React Native app that links all of the Expo SDK dependencies in the `packages/` directory in order to be able to run projects in the `apps/` directory. It currently only runs `test-suite` which is an Expo app with some custom code to turn it into a test runner. If you run `expo start` in the `test-suite` directory you can load the project in Expo Go. `bare-expo` imports the `test-suite` app root component and uses it as its own root component.
```
Comment (tsapeta) [inline CONTRIBUTING.md:25]: It's not only `test-suite` – `bare-expo` runs both `test-suite` and `native-component-list`.

Comment (vonovak) [reply CONTRIBUTING.md]: updated

**Review by tsapeta — APPROVED** [review]

**Comment by vonovak** [comment]
### Merge activity

* **Aug 14, 2:28 AM EDT**: @vonovak started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/expo/expo/30926).
